### PR TITLE
editoast: migrate rocket to actix

### LINKED ISSUE #3030: Migrate from Rocket to Actix Web
Motivations:

- Rocket is not maintain anymore
- Actix supports web sockets and is pretty much the most used framework now

### LINKED ISSUE #3067: Make editoast return `x-protobuf` for mvt tiles instead of `octet-stream`
### Description and goal

mvt tiles are protobuf so it would be cleaner to rteturn the exact type [here](https://github.com/DGEXSolutions/osrd/blob/dev/editoast/src/views/layers/layer.rs#L101).

:warning: should wait for actix https://github.com/DGEXSolutions/osrd/pull/3053 before doing this

### Acceptance criteria

check in the test that the correct type is returned

### PR DESCRIPTION
**Pull request by flomonster**

### CODE REVIEW

File: editoast/src/error.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+#[derive(Debug, Serialize)]
+pub struct InternalError {
+    #[serde(skip)]
+    status: StatusCode,
+    error_type: &'static str,
+    #[serde(flatten)]
+    exta: Map<String, Value>,
+    message: String,
+}
```
Comment (Khoyo) [inline editoast/src/error.rs]: s/exta/extra/
Also, #2697 defines the following format:

```
 "error": {
        "type": "no_path_found",
        "message": "No path could be found",
        "context": {}
    }
```

Shouldn't we start following that here? (Or should we update that format to the one used here - both options are fine by me)

File: editoast/src/map/mod.rs [hunk 2]
```diff
@@ -27,16 +27,16 @@ pub use self::redis_utils::{delete, get, keys, set};
 ///
 /// Returns the number of deleted keys
 async fn invalidate_full_layer_cache(
-    redis_pool: &RedisPool,
+    redis: &mut ConnectionManager,
```
Comment (Khoyo) [inline editoast/src/map/mod.rs:30]: s/redis/conn/ (or s/conn/redis/ everywhere else?)

File: editoast/src/views/layers/mod.rs [hunk 3]
```diff
@@ -1,10 +1,223 @@
...
+    }
+
+    #[actix_test]
+    async fn layer_view() {
+        test_get_query(
+            "/layers/layer/track_sections/mvt/does_not_exist?infra=2",
+            StatusCode::NOT_FOUND,
+            json!({
+                "message": "View 'does_not_exist' not found. Expected one of [\"geo\", \"sch\"]",
+                "error_type": "editoast:layers:ViewNotFound"
```
Comment (Khoyo) [inline editoast/src/views/layers/mod.rs]: Same rel. error format.

**Review by Khoyo — COMMENTED** [review]
Impressive.

File: editoast/src/error.rs (hunk 1 above)
Comment (flomonster) [reply editoast/src/error.rs]: Done. I did match the format we defined at #2697

File: editoast/src/error.rs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+impl InternalError {
+    pub fn get_type(&self) -> &'static str {
+        self.error_type
+    }
+
+    pub fn get_status(&self) -> StatusCode {
+        self.status
+    }
+}
```
Comment (Khoyo) [inline editoast/src/error.rs]: I think the `context()` overload is missing here

**Review by Khoyo — DISMISSED** [review]
Otherwise LGTM

Unless there are actix specific issues, but I couldn't tell either way ^^'

**Review by younesschrifi — APPROVED** [review]

**Review by Tguisnet — APPROVED** [review]

**Comment by Tguisnet** [comment]
I have few remarks on actix possibilities and features, but we'll discuss this later.

**Pull request merged**